src/main/java/com/puppycrawl/tools/checkstyle/api/ParserException.java

```diff
@@ -0,0 +1,43 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+/**
+ * Represents a java parser error.
+ *
+ * @author Pavel Bludov
+ * @noinspection CheckedExceptionClass
+ */
+public class ParserException extends CheckstyleException {
```

Comment:
for simple code moving .... new API class - it will never be ok.
API class it huge responsibility and contract with user code, reason for new API class should be very very good.

Reply:
It was my original intention that we use `java.text.ParseException` and not create a new one.

Reply:
`java.text.ParseException` has only one ctor:
```
public ParseException(String message, int offset)
```
but we need something like
```
public ParseException(String message, Throwable cause)
```